src/components/FieldInput.js

```diff
@@ -17,18 +17,22 @@ const useStyles = makeStyles({
 });
 
 export default function FieldInput(props) {
-  const { label, onChange, placeholder, variant } = props;
+  const { label, onChange, placeholder, variant, error = null } = props;
   const classes = useStyles();
 
   return (
     <div className={classes.root}>
       {label ? <h3 className={classes.label}>{label}</h3> : null}
+
       <TextField
         fullWidth
         margin="normal"
         variant={variant || 'standard'}
         placeholder={placeholder}
         onChange={onChange}
+        {...(error && { error: true, helperText: error })}
```

Comment:
do we need to shortcircuit this?

validateFunc gives '' = `false` if the field is valid, else a helper string which sets error to true. Also you already set helperText={error} below.